<a href="https://colab.research.google.com/github/lokeshkumarn/Notebooks/blob/master/BankTermDeposit.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import os
import numpy as np
import pandas as pd

#Visualizations
import matplotlib as mpl
from matplotlib import pyplot as plt

#Modeling
import sklearn

Load the Data

In [0]:
data = pd.read_excel('Bank Data for case study assignment.xlsx')

In [3]:
data.head()

age          job marital status   ... housing loan? Personal loan   y
0   30   unemployed         married  ...            no            no  no
1   33     services         married  ...           yes           yes  no
2   35   management          single  ...           yes            no  no
3   30   management         married  ...           yes           yes  no
4   59  blue-collar         married  ...           yes            no  no

[5 rows x 8 columns]

Exploratory Data Analysis and Statistical Analysis

In [20]:
data.columns

Index(['age', 'job', 'marital status ', 'education', 'credit default?',
       'housing loan?', 'Personal loan', 'y'],
      dtype='object')

Column Marital Status is having extra space in the end, just rename it without space in the end

In [0]:
data.rename(columns={'marital status ':'marital status'},inplace=True)

In [22]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1021 entries, 0 to 1020
Data columns (total 8 columns):
age                1021 non-null int64
job                1019 non-null object
marital status     1020 non-null object
education          1020 non-null object
credit default?    1020 non-null object
housing loan?      1019 non-null object
Personal loan      1019 non-null object
y                  1021 non-null object
dtypes: int64(1), object(7)
memory usage: 63.9+ KB


Total 1021 records in that Job, Marital Status, Education, Credit Default having 1 missing data and Housing Loan, Personal Loan has two missing data

First fill the missing the value for the job

In [23]:
data[data['job'].isnull()]

age  job marital status  ... housing loan? Personal loan    y
79     40  NaN        married  ...           yes            no   no
1004   42  NaN         single  ...           yes           yes  yes

[2 rows x 8 columns]

In [38]:
data[(data['age']== 40) & (data['marital status'] =='married') & (data['education'] =='secondary')
    & (data['credit default?'] =='no') & (data['housing loan?'] =='yes') & (data['Personal loan'] =='no')
    & (data['job'].isnull()==False)]['job']

270        services
482      technician
597      management
671      technician
684        services
710      technician
960      technician
1012    blue-collar
Name: job, dtype: object

Taking the Mode and imputing the missing value for job 

In [39]:
data[(data['age']== 40) & (data['marital status'] =='married') & (data['education'] =='secondary')
    & (data['credit default?'] =='no') & (data['housing loan?'] =='yes') & (data['Personal loan'] =='no')
    & (data['job'].isnull()==False)]['job'].mode()

0    technician
dtype: object

In [40]:
data[(data['age']== 40) & (data['marital status'] =='married') & (data['education'] =='secondary')
    & (data['credit default?'] =='no') & (data['housing loan?'] =='yes') & (data['Personal loan'] =='no')
    & (data['job'].isnull()==True)]

age  job marital status  ... housing loan? Personal loan   y
79   40  NaN        married  ...           yes            no  no

[1 rows x 8 columns]

imputing the new value for job using the mod value of job whose attributes had similar values as that of the missing values

In [48]:
data.loc[(data['age']== 40) & 
         (data['marital status'] =='married') & 
         (data['education'] =='secondary')    & 
         (data['credit default?'] =='no') & 
         (data['housing loan?'] =='yes') & 
         (data['Personal loan'] =='no')
    & (data['job'].isnull()==True),'job']

79    NaN
Name: job, dtype: object

In [0]:
data.loc[(data['age']== 40) & 
         (data['marital status'] =='married') & 
         (data['education'] =='secondary')    & 
         (data['credit default?'] =='no') & 
         (data['housing loan?'] =='yes') & 
         (data['Personal loan'] =='no')
    & (data['job'].isnull()==True),'job']=\
data[(data['age']== 40) & 
     (data['marital status'] =='married') & 
     (data['education'] =='secondary')    & 
     (data['credit default?'] =='no') & 
     (data['housing loan?'] =='yes') & 
     (data['Personal loan'] =='no')    & 
     (data['job'].isnull()==False)]['job'].mode()

In [47]:
data[data['job'].isnull()]

age  job marital status  ... housing loan? Personal loan    y
79     40  NaN        married  ...           yes            no   no
1004   42  NaN         single  ...           yes           yes  yes

[2 rows x 8 columns]